mlflow/utils/rest_utils.py

```diff
@@ -108,6 +108,21 @@ def http_request(
         ) from to
     except requests.exceptions.InvalidURL as iu:
         raise InvalidUrlException(f"Invalid url: {url}") from iu
+    except requests.exceptions.RetryError:
```

Comment:
This utility is used extensively in MLflow. Can we validate that this change has no adverse impacts to the client->tracking server in both OSS and Databricks and add client API tests that validate that there are no issues with current error responses associated with a retry-exhaustion failure mechanism for the tracking client?

Reply:
Thanks Ben! I had a offline discussion with @harupy and he suggested we could actually use `raise_on_status` field in the `Retry` config to handle this: https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html. The default value is `True` and I set it to `False` in gateway HTTP calls.

In this way, only the gateway caller behavior is changed while the other callers of `http_request` remain the same, thus the impact is minimized here.

Reply:
That is a FANTASTIC idea :D huge +1 to limiting the scope of the change only the gateway interface.